addons/io_hubs_addon/components/operators.py

```diff
@@ -670,6 +670,174 @@ def invoke(self, context, event):
+                            description="The pixel resoltion of the resulting lightmap.")
+    samples: IntProperty(name="Max Samples",
+                         default=1024,
+                         description="The number of samples to use for baking. Higher values reduce noise but take longer.")
+
+    def execute(self, context):
+        # Check selected objects
+        selected_objects = bpy.context.selected_objects
+
+        # filter mesh objects and others
+        mesh_objs, other_objs = [], []
+        for ob in selected_objects:
+            (mesh_objs if ob.type == 'MESH' else other_objs).append(ob)
+
+        for ob in other_objs:
+            # Remove non-mesh objects from selection to ensure baking will work
+            ob.select_set(False)
+
+        # set up UV layer structure. The first layer has to be UV0, the second one UV1 for the lightmap.
+        for obj in mesh_objs:
+            obj_uv_layers = obj.data.uv_layers
+            # Check whether there are any UV layers and if not, create the two that are required.
+            if len(obj_uv_layers) == 0:
+                obj_uv_layers.new(name='UV0')
+                obj_uv_layers.new(name='UV1')
+
+            # In case there is only one UV layer create a second one named "UV1" for the lightmap.
+            if len(obj_uv_layers) == 1:
+                obj_uv_layers.new(name='UV1')
+            # Check if object has a second UV layer. If it is named "UV1", assume it is used for the lightmap.
+            # Otherwise add a new UV layer "UV1" and place it second in the slot list.
+            elif obj_uv_layers[1].name != 'UV1':
+                print("The second UV layer in hubs should be named UV1 and is reserved for the lightmap, all the layers >1 are ignored.")
+                obj_uv_layers.new(name='UV1')
+                # The new layer is the last in the list, swap it for position 1
+                obj_uv_layers[1], obj_uv_layers[-1] = obj_uv_layers[-1], obj_uv_layers[1]
+
+            # The layer for the lightmap needs to be the active one before lightmap packing
+            obj_uv_layers.active = obj_uv_layers['UV1']
+
+        # run UV lightmap packing on all selected objects
+        bpy.ops.object.mode_set(mode='EDIT')
+        bpy.ops.mesh.select_all(action='SELECT')
+        # TODO: We need to warn the user at some place like the README that the uv_layer[1] gets completely overwritten if it is called 'UV1'
+        # bpy.ops.uv.lightmap_pack()
+        bpy.ops.uv.smart_project()
+        bpy.ops.object.mode_set(mode='OBJECT')
+        # Update the view layer so all parts take notice of the changed UV layout
+        bpy.context.view_layer.update()
+
+        # Gather all materials on the selected objects
+        materials = []
+        for obj in mesh_objs:
+            if len(obj.material_slots) >= 1:
+                # TODO: Make more efficient
+                for slot in obj.material_slots:
+                    if slot.material not in materials:
```

Comment:
I do so in line 737: if len(obj.material_slots) >= 1:

If there is no material present it does not bake the object but removes it from selection.

Reply:
The condition doesn't check if there are materials used. You may have one material slot but not assigned to material.
Why not just iterating over obj.data.materials here?
```
for obj in mesh_objs:
    for material in obj.data.materials:
        if material not in materials:
```
should work correctly.

Reply:
Ah actually it may not work, obj.data.materials may give an error
AttributeError: 'PointLight' object has no attribute 'materials'
I had that condition in my code
`if obj.type == 'MESH' and getattr(ob.data, "materials", None)`
so yeah probably do what @Exairnous suggested:

```
for obj in mesh_objs:
    for slot in obj.material_slots:
        if slot.material is not None and slot.material not in materials:
            materials.append(slot.material)
```

Reply:
Thank you, tested and it works! (thankfully I am already testing for mesh objects further up)